### ISSUE TITLE
Module name change

### ISSUE THREAD

**Issue by jacobwilliams** [jacobwilliams/json-fortran#33]
I'm thinking of changing the module name from json_module to jsonfortran (and the file to jsonfortran.f90).  This is just so they are consistent with the project name.  It will break backward compatibility, but I've already done that anyway with recent reorganization, so it shouldn't be a big deal.  Once this is done, the API should be pretty stable from here on, since I'm pretty happy with the all the nomenclature now.

**Comment by zbeekman** [comment]
I like some aspects of [Clerman and Spector’s](http://www.amazon.com/gp/product/052173052X/ref=as_li_tl?ie=UTF8&camp=1789&creative=390957&creativeASIN=052173052X&linkCode=as2&tag=izabbeeshom-20&linkId=KISKWXLNKLI44NNQ) naming scheme: Modules end in `_m` and derived types end in `_t`. Might be worthwhile to sleep on it, and make sure that you land on this as the final name. Changing these names will also cause the CMake build system to need to be updated.

**Comment by jacobwilliams** [comment]
I'm just going to leave it as is.
